File: libwebrtc-sys/src/video_track_source.cc
#include "libwebrtc-sys/include/webrtc_api.h"
#include "libwebrtc-sys/include/video_track_source.h"
#include "libwebrtc-sys/include/video_track_source_internal.h"

std::unique_ptr<ArcasVideoTrackSource> create_arcas_video_track_source()
{
    auto ref = rtc::make_ref_counted<ArcasVideoTrackSourceInternal>();
    return std::make_unique<ArcasVideoTrackSource>(ref);
}

File: libwebrtc-sys/src/video_track_source.rs
#[cxx::bridge]
pub mod ffi {
    unsafe extern "C++" {
        include!("include/alias.h");
        include!("include/video_track_source.h");
        type ArcasVideoTrackSource;
        type ArcasCxxVideoFrame = crate::video_frame::ffi::ArcasCxxVideoFrame;

        fn create_arcas_video_track_source() -> UniquePtr<ArcasVideoTrackSource>;
        // ArcasVideoTrackSource
        fn push_frame(self: &ArcasVideoTrackSource, video_frame: &ArcasCxxVideoFrame);
        fn cxx_clone(self: &ArcasVideoTrackSource) -> UniquePtr<ArcasVideoTrackSource>;
    }

    extern "Rust" {}
}
